lib/streamlit/testing/v1/app_test.py

```diff
@@ -0,0 +1,266 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import hashlib
+import pathlib
+import tempfile
+import textwrap
+import traceback
+from typing import Any, Sequence
+from unittest.mock import MagicMock
+
+from streamlit import source_util
+from streamlit.proto.WidgetStates_pb2 import WidgetStates
+from streamlit.runtime import Runtime
+from streamlit.runtime.caching.storage.dummy_cache_storage import (
+    MemoryCacheStorageManager,
+)
+from streamlit.runtime.media_file_manager import MediaFileManager
+from streamlit.runtime.memory_media_file_storage import MemoryMediaFileStorage
+from streamlit.runtime.state.session_state import SessionState
+from streamlit.testing.v1.element_tree import (
+    Block,
+    Button,
+    Caption,
+    Checkbox,
+    Code,
+    ColorPicker,
+    DateInput,
+    Divider,
+    ElementList,
+    ElementTree,
+    Exception,
+    Header,
+    Latex,
+    Markdown,
+    Multiselect,
+    Node,
+    NumberInput,
+    Radio,
+    Selectbox,
+    SelectSlider,
+    Slider,
+    Subheader,
+    Text,
+    TextArea,
+    TextInput,
+    TimeInput,
+    Title,
+    WidgetList,
+)
+from streamlit.testing.v1.local_script_runner import LocalScriptRunner
+from streamlit.testing.v1.util import patch_config_options
+
+TMP_DIR = tempfile.TemporaryDirectory()
+
+
+class AppTest:
```

Comment:
Could we extend AppTest somehow from ElementTree so that we don't have to duplicate all command properties?

Reply:
We could, though it might make the implementation more confusing